src/nedap/utils/test/impl.cljc

```diff
@@ -64,22 +64,32 @@
      (defmethod cljs.test/report [::exit-code-reporter :end-run-tests] [summary]
        (set-exit-code-for! summary js/process))))
 
+(defmulti expect-matcher
+  "Given a symbol + ns returns :assert-expr and :pred.
+
+  `#'expect` uses `pred` is to compare values in assertions: `(not (pred from, to)`, `(pred from to-change).
+  assert-expr is the symbol used to construct the `is` form, which allows error-reporting such as "
+  (fn [sym _ns] sym))
+
+(defmethod expect-matcher :default [sym ns] {:assert-expr sym :pred (ns-resolve ns sym)})
+
 (defn expect
-  [bodies {:keys [to-change from to] :as opts} clj?]
+  [bodies {:keys [to-change from to with] :as opts} clj? ns]
   {:pre [(spec/valid? boolean? clj?)]}
   (assert (seq bodies) "bodies can't be empty")
-  (assert (= #{:to-change :from :to} (set (keys opts))) (pr-str opts))
-  (assert (not (meta= [from to]))
-          (binding [*print-meta* true]
-            (str (pr-str from) " should be different from " (pr-str to))))
-  (assert (some? to-change) (pr-str to-change))
+  (assert (= #{:to-change :from :to} (set (keys (dissoc opts :with)))) (pr-str opts))
+  (let [{:keys [pred assert-expr]} (expect-matcher with ns)
```

Comment:
Seems good to assert `pred assert-expr` at macroexpasion-time, as having the error at runtime would be much more confusing